File: problems/Binary Tree Level Order Traversal/levelOrder.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        var bfs = new Queue<TreeNode>();
        var result = new List<IList<int>>();

        if (null != root) bfs.Enqueue(root);

        while (0 < bfs.Count) {
            var count = bfs.Count;

            result.Add(new List<int>());

            while (0 < count--) {
                var node = bfs.Dequeue();

                if (null != node) {
                    result[result.Count - 1].Add(node.val);

                    if (null != node.left) bfs.Enqueue(node.left);
                    if (null != node.right) bfs.Enqueue(node.right);
                }
            }
        }

        return result;
    }
}

File: problems/N-ary Tree Level Order Traversal/levelOrder.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> result = new ArrayList();
        Queue<Node> bfs = new LinkedList<Node>();

        if (null != root) {
            bfs.add(root);
        }

        while (!bfs.isEmpty()) {
            int size = bfs.size();
            List<Integer> level = new ArrayList();

            for (int idx = 0; size > idx; ++idx) {
                Node node = bfs.poll();

                level.add(node.val);

                for (Node child : node.children) {
                    bfs.add(child);
                }
            }

            result.add(level);
        }

        return result;
    }
}
